### PR TITLE
APEXCORE-619 Checkpoint leaf stateless operator for correct recovery after application restart.

### PR DESCRIPTION
**Pull request by tushargosavi**

### CODE REVIEW

**Comment by sandeshh** [comment]
Purging of checkpoints skips the stateless operators, so please check that purging is happening correctly.

File: common/src/main/java/com/datatorrent/common/util/LeafStatelessStorageAgent.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+   * @param windowId
+   * @throws IOException
+   */
+  @Override
+  public void save(Object object, int operatorId, long windowId) throws IOException
+  {
+    if (windowId == Stateless.WINDOW_ID) {
+      super.save(object, operatorId, windowId);
+    } else {
+      FSDataOutputStream fout = null;
```
Comment (sandeshh) [inline common/src/main/java/com/datatorrent/common/util/LeafStatelessStorageAgent.java:64]: try with resources?

**Comment by tushargosavi** [comment]
our check-style rules fails the build for empty block, hence kept it as it is.

**Comment by tushargosavi** [comment]
purging checkpoint is working as LeafStatelessStorageAgent is extended from FSStatelessStorageAgent

**Comment by pramodin** [comment]
@tushargosavi It's been a while so can you rebase and I can look into it

**Comment by vrozov** [comment]
@tushargosavi @PramodSSImmaneni closed due to inactivity.

**Pull request closed without merging**